proposals/PROPOSALS.md

```diff
@@ -0,0 +1,158 @@
+
+The aim of the Haskell Improvement Process is to apply the openness and collaboration that have shaped Haskell's documentation and implementation to the process of evolving the language and its ecosystem. This document captures our guidelines, commitments and expectations regarding this process.
+
+## Why Write a Proposal?
+
+HIPs are key to making Haskell better for the good of everyone. If you decide to invest the time and effort of putting a proposal forward and seeing it through, your efforts and time will shape and improve the Haskell ecosystem, which means that your proposal may impact the life of a myriad of developers all over the world, including those on your own team. For many, this aspect alone can be quite worthwhile.
+
+However, it's important to note that seeing a proposal through to its conclusion is an involved task. On the one hand, it takes time to convince people that your suggestions are a worthwhile change for hundreds of thousands of developers to accept. Particularly given the sheer volume of developers that could be affected by a proposal, its acceptance is conservative and carefully thought through. Typically, this includes many rounds of discussion with HF leaders, Haskell library maintainers, and the overall community, several iterations on the design of the proposal, and some effort at prototyping the proposed change. Often, it takes weeks to months of discussion, re-design, and prototyping for a proposal to be accepted. It is therefore important to note that seeing a proposal through to its conclusion can be time-consuming and not all proposals may end up accepted, although they may teach us all something!
+
+If you’re motivated enough to go through this involved but rewarding process, go on with writing and keep on reading.
+
+## What's the process for submitting a HIP?
+
+There are four major steps in the HIP process:
+
+1. Initial informal discussion (2 weeks)
+2. Submission
+3. Formal presentation (up to 1 month)
+4. Formal evaluation (up to 2 months)
+
+### Initial informal discussion (2 weeks)
+
+Before submitting a HIP, it is required that you perform necessary preparations:
+
+- Discuss your idea on the [Haskell.org Discourse](https://discourse.haskell.org/)  currently, we suggest cross-posting on the Haskell Foundation Slack for higher volume commentary) Create a Haskell Foundation topic that starts with “Pre-HIP” and briefly describe what you would like to change and why you think it’s a good idea.
+
+- Proposing your ideas on the Discourse is not an optional step. For every change to the ecosystem, it is important to *publicly* gauge interest from the community and relevant stakeholders. Use this step to promote your idea and gather early feedback on your informal proposal. It may happen that experts and community members may have tried something similar in the past and may offer valuable advice.
```

Comment:
From @goldfirere: 

> But the proposal process itself is public, no?

Reply:
From @myShoggoth 

> can

(re: "may")

Reply:
This one is tricky, because it's both a stopgap against people not doing their due diligence, but also to make sure Pre-HIPs are circulated as widely as is reasonable possible. However, it *is* a slightly optional step in my mind: the proposal process does indeed handle the "public" part of this. Perhaps this should be an *internal* policy, to make sure that anything we propose goes through this informal due diligence to make sure we're not stepping on anyone's toes. It may be worth specifying additional criteria for how proposals arise from within the HF.